planner/core/src/main/java/eu/seaclouds/platform/planner/core/Planner.java

```diff
@@ -167,17 +174,19 @@ public String rePlan() throws ParsingException, IOException {
         List<NameValuePair> optParams = getReqParams(
                 new BasicNameValuePair("aam", aam),
                 new BasicNameValuePair("dam", dam),
-                new BasicNameValuePair("mmResults", mmOutput)
+                new BasicNameValuePair("offers", mmOutput)
         );
-        String optResult = optimizerClient.postRequest("/optimize", optParams);
-        return optResult; //TODO: check the output for the optimizer
+
+        Pair<String, String> optResult = optimizerClient.postRequest("/optimize", optParams);
+        if(optResult.first.equals("200")){
+            return optResult.second; //TODO: check the output for the optimizer
+        }
+        return null; //TODO: return failure
     }
 
     public String generateDam() {
```

Comment:
The optimizer returns a JSon string that I must forward to the service caller. So String is more appropriate than a class to be serialized again.